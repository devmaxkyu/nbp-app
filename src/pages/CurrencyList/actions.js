/** 
 * @author Zemin W.
 * @description define actions to refresh currencies list from nbp-api
 * @lastModified 2019.10.4
 * */

const REFRESH_LIST = 'LIST_UPDATE'

const FILTER_LIST = 'SET_FILTER'

export const actions = {
    REFRESH_LIST,
    FILTER_LIST
}

export const refresh = data =>({
    type: REFRESH_LIST,
    data
})

export const filter = data =>({
    type: FILTER_LIST,
    data
})