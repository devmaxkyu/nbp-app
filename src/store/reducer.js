/** 
 * @author Zemin W.
 * @description combine reducers
 * @lastModified 2019.10.4
 * */

import { combineReducers } from 'redux'

import CurrencyDetailReducer from '../pages/CurrencyDetail/reducer'
import CurrencyListReducer from '../pages/CurrencyList/reducer'
import FavouriteReducer from '../pages/Favourite/reducer'

export default combineReducers({
    CurrencyDetailReducer,
    CurrencyListReducer,
    FavouriteReducer
})
