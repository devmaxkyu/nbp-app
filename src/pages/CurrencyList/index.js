/** 
 * @description page component to show currencies list
 * @lastModified 2019.10.4
 * @author zemin.
 * */

import React from 'react'
import { connect } from 'react-redux'
import {refresh, table} from './actions'
import {getCurrencyList, TABLE_FORMAT} from '../../services/nbpApi'
import Table from '../../components/Table/currency'
import ErrorBoundary from '../../components/Error'


class CurrencyList extends React.Component {
    constructor(props){
        super(props)        
    }

    componentDidMount(){
        if(!this.props.currencies.length){                        
            getCurrencyList(this.props.table, this.setCurrencies)            
        }
    }

    setCurrencies = (result) => {
        const {dispatch} = this.props
        //console.log('CurrencyList.index.setCurrencies',result);
        
        dispatch(refresh(result[0].rates))        
    }

    
    handleTableA = () => {
        const {dispatch} = this.props
        dispatch(table(TABLE_FORMAT.A))        
        getCurrencyList(TABLE_FORMAT.A, this.setCurrencies)        
    }

    handleTableB = () => {
        const {dispatch} = this.props
        dispatch(table(TABLE_FORMAT.B))
        getCurrencyList(TABLE_FORMAT.B, this.setCurrencies)
    }

    handleTableC = () => {
        const {dispatch} = this.props
        dispatch(table(TABLE_FORMAT.C))
        getCurrencyList(TABLE_FORMAT.C, this.setCurrencies)
    }
    render() {
        
        return (
            <div>
                <h3>Currency List</h3>
                <div className="btn-group">
                    
                        <button className={"App-link "+(this.props.table==TABLE_FORMAT.A?"active":"")} name="table_A" onClick={this.handleTableA}>
                            Table {TABLE_FORMAT.A}                                                        
                        </button>
                        <button className={"App-link "+(this.props.table==TABLE_FORMAT.B?"active":"")} name="table_B" onClick={this.handleTableB}>
                            Table {TABLE_FORMAT.B}                                                        
                        </button>
                        {/* <button className={this.state.table==TABLE_FORMAT.C?"":"App-btn App-btn-active"} name="table_C" onClick={this.handleTableC}>
                            Table {TABLE_FORMAT.C}                                                        
                        </button> */}
                    
                </div>                
                <div className="table-wrap">
                    <ErrorBoundary>
                        <Table items = {this.props.currencies} table = {this.props.table} />
                    </ErrorBoundary>
                </div>
                
            </div>
        )
    }
}


export default connect(state => ({
    currencies: state.currencyList.list,
    table: state.currencyList.table
}))(CurrencyList)