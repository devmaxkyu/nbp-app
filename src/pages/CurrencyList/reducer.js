/** 
 * @author Zemin W.
 * @description define reducers to refresh and filter currencies list
 * @lastModified 2019.10.4
 * */
import {createReducer} from 'redux-create-reducer'
import {actions} from './actions'

 const initialState = {
     list: [],
 }

 const reducer = createReducer(initialState, {
     [actions.REFRESH_LIST] (state, action){
         return { 
             ...state,
             list: action.data
         }
     },

     [actions.FILTER_LIST] (state, action){
        return { 
            ...state,
            filter: action.data
        }
    },
 })

 export default reducer
