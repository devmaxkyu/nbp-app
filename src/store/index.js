/** 
 * @author Zemin W.
 * @description create store object
 * @lastModified 2019.10.4
 * */

import { createStore, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import reducer from './reducer'

const store = createStore(reducer, applyMiddleware(createLogger()))

export default store
