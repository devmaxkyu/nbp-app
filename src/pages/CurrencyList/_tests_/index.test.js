
import React from 'react'
import CurrencyList from '../../CurrencyList'
import renderer from 'react-test-renderer'
import {Provider} from 'react-redux'
import store from '../../../store'

import { BrowserRouter as Router} from 'react-router-dom'

jest.mock('../../../services/nbpApi')

it('Currency List rendering', () => {
  const component = renderer.create(
    <Provider store = {store}><Router><CurrencyList/></Router></Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});