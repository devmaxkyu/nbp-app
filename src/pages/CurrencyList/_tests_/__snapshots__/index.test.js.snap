// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency List rendering 1`] = `
<div>
  <h3>
    Currency List
  </h3>
  <div
    className="btn-group"
  >
    <button
      className="App-link active"
      name="table_A"
      onClick={[Function]}
    >
      Table 
      A
    </button>
    <button
      className="App-link "
      name="table_B"
      onClick={[Function]}
    >
      Table 
      B
    </button>
  </div>
  <div
    className="table-wrap"
  >
    <table
      className="App-table"
    >
      <thead>
        <tr>
          <th>
            Currency Name
          </th>
          <th>
            Currency Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          id="THB"
        >
          <td>
            <a
              href="/currency/A/THB"
              onClick={[Function]}
            >
              bat (Tajlandia)
            </a>
          </td>
          <td>
            0.1289
          </td>
        </tr>
        <tr
          id="USD"
        >
          <td>
            <a
              href="/currency/A/USD"
              onClick={[Function]}
            >
              dolar amerykański
            </a>
          </td>
          <td>
            3.9145
          </td>
        </tr>
        <tr
          id="AUD"
        >
          <td>
            <a
              href="/currency/A/AUD"
              onClick={[Function]}
            >
              dolar australijski
            </a>
          </td>
          <td>
            2.6589
          </td>
        </tr>
        <tr
          id="HKD"
        >
          <td>
            <a
              href="/currency/A/HKD"
              onClick={[Function]}
            >
              dolar Hongkongu
            </a>
          </td>
          <td>
            0.4993
          </td>
        </tr>
        <tr
          id="CAD"
        >
          <td>
            <a
              href="/currency/A/CAD"
              onClick={[Function]}
            >
              dolar kanadyjski
            </a>
          </td>
          <td>
            2.9487
          </td>
        </tr>
        <tr
          id="NZD"
        >
          <td>
            <a
              href="/currency/A/NZD"
              onClick={[Function]}
            >
              dolar nowozelandzki
            </a>
          </td>
          <td>
            2.4808
          </td>
        </tr>
        <tr
          id="SGD"
        >
          <td>
            <a
              href="/currency/A/SGD"
              onClick={[Function]}
            >
              dolar singapurski
            </a>
          </td>
          <td>
            2.8487
          </td>
        </tr>
        <tr
          id="EUR"
        >
          <td>
            <a
              href="/currency/A/EUR"
              onClick={[Function]}
            >
              euro
            </a>
          </td>
          <td>
            4.3097
          </td>
        </tr>
        <tr
          id="HUF"
        >
          <td>
            <a
              href="/currency/A/HUF"
              onClick={[Function]}
            >
              forint (Węgry)
            </a>
          </td>
          <td>
            0.012986
          </td>
        </tr>
        <tr
          id="CHF"
        >
          <td>
            <a
              href="/currency/A/CHF"
              onClick={[Function]}
            >
              frank szwajcarski
            </a>
          </td>
          <td>
            3.9192
          </td>
        </tr>
        <tr
          id="GBP"
        >
          <td>
            <a
              href="/currency/A/GBP"
              onClick={[Function]}
            >
              funt szterling
            </a>
          </td>
          <td>
            4.8929
          </td>
        </tr>
        <tr
          id="UAH"
        >
          <td>
            <a
              href="/currency/A/UAH"
              onClick={[Function]}
            >
              hrywna (Ukraina)
            </a>
          </td>
          <td>
            0.1592
          </td>
        </tr>
        <tr
          id="JPY"
        >
          <td>
            <a
              href="/currency/A/JPY"
              onClick={[Function]}
            >
              jen (Japonia)
            </a>
          </td>
          <td>
            0.036246
          </td>
        </tr>
        <tr
          id="CZK"
        >
          <td>
            <a
              href="/currency/A/CZK"
              onClick={[Function]}
            >
              korona czeska
            </a>
          </td>
          <td>
            0.1668
          </td>
        </tr>
        <tr
          id="DKK"
        >
          <td>
            <a
              href="/currency/A/DKK"
              onClick={[Function]}
            >
              korona duńska
            </a>
          </td>
          <td>
            0.577
          </td>
        </tr>
        <tr
          id="ISK"
        >
          <td>
            <a
              href="/currency/A/ISK"
              onClick={[Function]}
            >
              korona islandzka
            </a>
          </td>
          <td>
            0.031298
          </td>
        </tr>
        <tr
          id="NOK"
        >
          <td>
            <a
              href="/currency/A/NOK"
              onClick={[Function]}
            >
              korona norweska
            </a>
          </td>
          <td>
            0.4299
          </td>
        </tr>
        <tr
          id="SEK"
        >
          <td>
            <a
              href="/currency/A/SEK"
              onClick={[Function]}
            >
              korona szwedzka
            </a>
          </td>
          <td>
            0.3985
          </td>
        </tr>
        <tr
          id="HRK"
        >
          <td>
            <a
              href="/currency/A/HRK"
              onClick={[Function]}
            >
              kuna (Chorwacja)
            </a>
          </td>
          <td>
            0.5803
          </td>
        </tr>
        <tr
          id="RON"
        >
          <td>
            <a
              href="/currency/A/RON"
              onClick={[Function]}
            >
              lej rumuński
            </a>
          </td>
          <td>
            0.9055
          </td>
        </tr>
        <tr
          id="BGN"
        >
          <td>
            <a
              href="/currency/A/BGN"
              onClick={[Function]}
            >
              lew (Bułgaria)
            </a>
          </td>
          <td>
            2.2035
          </td>
        </tr>
        <tr
          id="TRY"
        >
          <td>
            <a
              href="/currency/A/TRY"
              onClick={[Function]}
            >
              lira turecka
            </a>
          </td>
          <td>
            0.6695
          </td>
        </tr>
        <tr
          id="ILS"
        >
          <td>
            <a
              href="/currency/A/ILS"
              onClick={[Function]}
            >
              nowy izraelski szekel
            </a>
          </td>
          <td>
            1.1151
          </td>
        </tr>
        <tr
          id="CLP"
        >
          <td>
            <a
              href="/currency/A/CLP"
              onClick={[Function]}
            >
              peso chilijskie
            </a>
          </td>
          <td>
            0.005444
          </td>
        </tr>
        <tr
          id="PHP"
        >
          <td>
            <a
              href="/currency/A/PHP"
              onClick={[Function]}
            >
              peso filipińskie
            </a>
          </td>
          <td>
            0.0759
          </td>
        </tr>
        <tr
          id="MXN"
        >
          <td>
            <a
              href="/currency/A/MXN"
              onClick={[Function]}
            >
              peso meksykańskie
            </a>
          </td>
          <td>
            0.2016
          </td>
        </tr>
        <tr
          id="ZAR"
        >
          <td>
            <a
              href="/currency/A/ZAR"
              onClick={[Function]}
            >
              rand (Republika Południowej Afryki)
            </a>
          </td>
          <td>
            0.2622
          </td>
        </tr>
        <tr
          id="BRL"
        >
          <td>
            <a
              href="/currency/A/BRL"
              onClick={[Function]}
            >
              real (Brazylia)
            </a>
          </td>
          <td>
            0.9524
          </td>
        </tr>
        <tr
          id="MYR"
        >
          <td>
            <a
              href="/currency/A/MYR"
              onClick={[Function]}
            >
              ringgit (Malezja)
            </a>
          </td>
          <td>
            0.9348
          </td>
        </tr>
        <tr
          id="RUB"
        >
          <td>
            <a
              href="/currency/A/RUB"
              onClick={[Function]}
            >
              rubel rosyjski
            </a>
          </td>
          <td>
            0.061
          </td>
        </tr>
        <tr
          id="IDR"
        >
          <td>
            <a
              href="/currency/A/IDR"
              onClick={[Function]}
            >
              rupia indonezyjska
            </a>
          </td>
          <td>
            0.00027708
          </td>
        </tr>
        <tr
          id="INR"
        >
          <td>
            <a
              href="/currency/A/INR"
              onClick={[Function]}
            >
              rupia indyjska
            </a>
          </td>
          <td>
            0.055147
          </td>
        </tr>
        <tr
          id="KRW"
        >
          <td>
            <a
              href="/currency/A/KRW"
              onClick={[Function]}
            >
              won południowokoreański
            </a>
          </td>
          <td>
            0.003304
          </td>
        </tr>
        <tr
          id="CNY"
        >
          <td>
            <a
              href="/currency/A/CNY"
              onClick={[Function]}
            >
              yuan renminbi (Chiny)
            </a>
          </td>
          <td>
            0.5514
          </td>
        </tr>
        <tr
          id="XDR"
        >
          <td>
            <a
              href="/currency/A/XDR"
              onClick={[Function]}
            >
              SDR (MFW)
            </a>
          </td>
          <td>
            5.3444
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
